### PR TITLE
adapt makefiles to fix parallel make

### PR DESCRIPTION
**Pull request by tukss**
This introduces a dependency on the `makedepf90` tool to automatically
create the module dependencies.

`Legendre_Transforms_newlayout.F90` was deleted because it also provides
the `Legendre_Transforms` module leading to build errors.

### CODE REVIEW

**Comment by tukss** [comment]
I deleted `Legendre_Transforms_newlayout.F90` to fix build errors. It seems that it was not used before, but someone more knowledgeable should comment.

**Comment by tukss** [comment]
This introduces `makedepf90` as a dependency. This is a small tool written in C to generate the F90 dependency tree. If we don't want this as a dependency we could instead just commit the file `Makefile.fdeps` and update it manually every time the module structure changes. A third option might be to do this automatically with a hook.

**Comment by gassmoeller** [comment]
Nice work, I like it. I tested this PR and it works for me. However, I think adding a new dependency is something we might not want, and @feathern should decide about. Another option is to check in the dependency file, and let the tester check for changes with every PR and report back when we need to update the dependency file.

**Comment by tukss** [comment]
> Another option is to check in the dependency file, and let the tester check for changes with every PR and report back when we need to update the dependency file.

Yes, that sounds like a good compromise. The dependency file should not change that often anyway. We could add optional make target to recreate the dependency file.

**Comment by tukss** [comment]
> I deleted `Legendre_Transforms_newlayout.F90` to fix build errors. It seems that it was not used before, but someone more knowledgeable should comment.

@feathern can you quickly comment on whether we need to keep the `Legendre_Transforms_newlayout.F90` file?

File: src/Makefile [hunk 1]
```diff
@@ -7,11 +7,11 @@ include object_list  #List of object files and associated variables
 
 ######################################################
 
-MAKENP  = $(MAKE) --no-print-directory
+MAKEFLAGSS  += --no-print-directory
```
Comment (gassmoeller) [inline src/Makefile]: Is it `MAKEFLAGS` instead of `MAKEFLAGSS`? I have to little experience with make to know if this is correct.

Comment (tukss) [reply src/Makefile]: Good point. That is a typo. Actually `--no-print-directory` is already passed down from the top `make` invocation, so I'm just removing that line.

**Pull request merged**